src/jit/morph.cpp

```diff
@@ -2301,13 +2300,15 @@ void fgArgInfo::EvalArgsToTemps()
             // Create a placeholder node to put in its place in gtCallLateArgs.
 
             // For a struct type we also need to record the class handle of the arg.
-            CORINFO_CLASS_HANDLE clsHnd = NULL;
+            CORINFO_CLASS_HANDLE clsHnd = NO_CLASS_HANDLE;
 
 #if defined(_TARGET_AMD64_) && !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
 
+            // All structs are either passed (and retyped) as integral types, OR they
+            // are passed by reference.
             noway_assert(argx->gtType != TYP_STRUCT);
 
-#else // !(defined(_TARGET_AMD64_) && !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING))
+#else // !(defined(_TARGET_AMD64_) || defined(FEATURE_UNIX_AMD64_STRUCT_PASSING))
```

Comment:
I didn't do it correctly, but what I intended to change it to was:
!defined(_TARGET_AMD64_) || defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
(i.e. eliminating the outer braces and distributing the '!' because I find it too easy to miss that the '!' applies to the whole expression.) Would you prefer the former?